jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -681,6 +685,13 @@ def apply_RGB_presets(self):
     def vue_apply_RGB_presets(self, data):
         self.apply_RGB_presets()
 
+    def _update_stretch_on_slice_change(self, msg):
+        print(msg.slice)
+        viewer = self.viewer.selected_obj[0] if self.viewer_multiselect else self.viewer.selected_obj  # noqa
+        sliced_data = viewer.state.reference_data.get_object(statistic=None)[:, :, msg.slice]

```

Comment:
Not idea, this is going to pull the whole cube into memory first.

Reply:
Definitely not ideal, just wanted to see if it was possible. Ideal will be for the developer that pulls this ticket!